code/modules/projectiles/guns/specialist.dm

```diff
@@ -721,6 +721,66 @@
 		if(!auto_fire)
 			STOP_PROCESSING(SSobj, src)
 
+//CO SMARTGUN
+/obj/item/weapon/gun/smartgun/co
+	name = "\improper M56C 'Command' smartgun"
+	desc = "The actual firearm in the 4-piece M56C Smartgun system. Back order only. Besides a more robust weapons casing, an ID lock system and a fancy paintjob, the gun's performance is identical to the standard-issue M56B."
+	icon_state = "m56c"
+	item_state = "m56c"
+	flags_gun_features = GUN_AUTO_EJECTOR|GUN_SPECIALIST|GUN_WIELDED_FIRING_ONLY|GUN_HAS_FULL_AUTO
+
+	var/mob/living/carbon/human/linked_human
+	var/is_locked = TRUE
+
+/obj/item/weapon/gun/smartgun/co/able_to_fire(mob/user)
+	. = ..()
+	if(is_locked && linked_human && linked_human != user)
+		if(linked_human.is_revivable() || linked_human.stat != DEAD)
+			to_chat(user, SPAN_WARNING("[icon2html(src)] Trigger locked by [src]. Unauthorized user."))
+			playsound(loc,'sound/weapons/gun_empty.ogg', 25, 1)
+			return FALSE
+
+		linked_human = null
+		is_locked = FALSE
+		UnregisterSignal(linked_human, COMSIG_PARENT_QDELETING)
+
+/obj/item/weapon/gun/smartgun/co/pickup(user)
+	if(!linked_human)
+		src.name_after_co(user, src)
+		to_chat(usr, SPAN_NOTICE("[icon2html(src)] You pick up [src], registering yourself as its owner."))
+	..()
+
+/obj/item/weapon/gun/smartgun/co/verb/toggle_lock()
+	set category = "Weapons"
+	set name = "Toggle Lock"
+	set src in usr
+
+	if(usr != linked_human)
+		to_chat(usr, SPAN_WARNING("[icon2html(src)] Action denied by [src]. Unauthorized user."))
+		return
+
+	is_locked = !is_locked
+	to_chat(usr, SPAN_NOTICE("[icon2html(src)] You [is_locked? "lock": "unlock"] [src]."))
+	playsound(loc,'sound/machines/click.ogg', 25, 1)
+
+/obj/item/weapon/gun/smartgun/co/proc/name_after_co(var/mob/living/carbon/human/H, var/obj/item/weapon/gun/smartgun/co/I)
```

Comment:
confusing proc name

Reply:
Yell at whoever made the original proc, I copy pasted this from the M46C code.